file/writer.go

```diff
@@ -360,6 +360,16 @@ func KongStateToFile(kongState *state.KongState, config WriteConfig) error {
 			zeroOutTimestamps(k)
 			c.ACLGroups = append(c.ACLGroups, &k.ACLGroup)
 		}
+		mtlsAuths, err := kongState.MTLSAuths.GetAllByConsumerID(*c.ID)
+		if err != nil {
+			return err
+		}
+		for _, k := range mtlsAuths {
+			zeroOutID(k, nil, config.WithID)
```

Comment:
This is a little complicated but I think we discussed that ID will always be written out and not be a conditional thing because ID is the only unique field for this entity.